### ISSUE TITLE
Feature Request: Linked Domains

### ISSUE THREAD

**Issue by k44me** [stoically/temporary-containers#499]
Google now forwards its privacy consent for youtube.com to consent.google.com... Ergo, for temporary containers to be able to handle youtube in a meaningful way, it must be possible to connect domains. This is true for other sites too, who use some authentication mechanism spanning multiple domains.

What I mean by this, is that I want a configuration option, that handles things differently than isolation:"different from tab domain and sub-domain", but instead goes with "connected domains". E.g. if I open youtube, and it forwards me to consent.google.com, navigation should not go to a new container, but remain in the same container.

E.g. I want to specify that "youtube.com" may forward to "google.com" (and consequently also return from "google.com") without entering a new container. But every other forwarding to any other domain should open a new container.

And this behaviour should of course happen in every temporary container. Because the whole point of using temporary containers is that google cannot track me. So I definitely do not want to create a permanent container for youtube where I play all youtube videos...

**Comment by crssi** [comment]
I have same problems how to solve the G shit.
You can't even block `googleads.g.doubleclick.net` anymore to get though. 😢

**Comment by Jakeler** [comment]
Doesn't this already work with the per domain settings? This is my config:

Domain Pattern:
`consent.youtube.com`

Exclude Target Domains:
`*.youtube.com`
`consent.google.*`
`googleads.g.doubleclick.net`

So a tab with `consent.youtube.com` can forward to other youtube subdomains, google and doubleclick targets without creating new temporary containers. Note that it excludes google.*, because it sometimes hits other TLDs (.de in my case) as well.

**Comment by AdKiller** [comment]
@ Jakeler Thanks for the tip! this fixed the new Google Consent shit for me!
It's been years that I have not configured Temporary Containers and I had forgotten how.

**Comment by k44me** [comment]
> Doesn't this already work with the per domain settings? This is my config:
> 
> Domain Pattern:
> `consent.youtube.com`

Yes and no. I do NOT want per-domain isolation (or am I misinterpreting what that really is?). I want global isolation. Meaning that if I open a url to youtube from any site, I want a completely new tab isolated from all previous youtube tabs. But I want to allow navigation inside this youtube tab (and every other youtube tab) to consent.google.com, without any of the tabs knowing anything about each other...

**Comment by Jakeler** [comment]
@k44me "Global" and "per domain" are just categories in the settings, not different isolation modes. They work together, so you can configure the global (common) behavior and then override some settings for a specific domain. I think what you describe is exactly what I want as well and the my config should work for you too, just try it out.

@AdKiller Yep, I haven't checked Maps so far... Now I added another Domain Pattern for `consent.google.*` with exclude for `*.google.*` in the hope that this finally handles all google services.

**Comment by AdKiller** [comment]
@Jakeler thanks for the suggestion but seems like currently it is impossible to use Google Maps and Automatic Mode enabled.

**Comment by Jakeler** [comment]
@AdKiller It works for me even with Automatic Mode, both from new tabs and google search results. Domain pattern `consent.google.*` needs also youtube and doubleclick exludes, because they always try to send the result to all domains... maybe that was not clear in my last comment. 
If it still does not work you can check with Firefox dev tools - [network monitor](https://developer.mozilla.org/en-US/docs/Tools/Network_Monitor) what domains are missing, they show up in red "blocked by Temporary Containers".

**Comment by AdKiller** [comment]
@Jakeler thanks for the tip about network monitor! I had *.youtube.com in  Exclude Target Domains but not consent.youtube.com. adding that seems to fix the problem. what a pain in the ass!

**Comment by JonnyCodewalker** [comment]
@AdKiller sorry to bother, but can you write a list of what exact exclusions I have to write to get youtube to not loop endlessly? I tried to use the network manager to find out, but because of the loop that didn't really work.

**Comment by AdKiller** [comment]
@JonnyCodewalker 

Exclude Target Domains:
*.youtube.com
*.google.*
googleads.g.doubleclick.net
consent.youtube.com 


 Ignoring requests to:
 addons.mozilla.org
consent.youtube.com
consent.google.*
maps.google.*
googleads.g.doubleclick.net 

Automatic Mode: enabled
